build.sbt

```diff
@@ -295,11 +297,22 @@ val sttpV = "2.2.9"
 //we use legacy version because this one supports Scala 2.12
 val monocleV = "2.1.0"
 
-lazy val commonDockerSettings = {
+lazy val commonDockerSettings = (workingDir: String) => {
   Seq(
     dockerBaseImage := "openjdk:11-jre-slim",
     dockerUsername := dockerUserName,
     dockerUpdateLatest := dockerUpLatestFromProp.getOrElse(!isSnapshot.value),
+    dockerCommands := {
+      val (before, after) = dockerCommands.value.splitAt(dockerCommands.value.size - 3) // 3rd from the end command is setting USER, and we want to do the following yet as root
```

Comment:
Let's do it via dist, so it doesn't clutter docker and is available also in tar distribution. E.g via https://github.com/sbt/sbt-javaagent ?